src/odemis/driver/simcam.py

```diff
@@ -58,7 +58,7 @@ def __init__(self, name, role, image, children=None, daemon=None, **kwargs):
         converter = dataio.find_fittest_converter(image, mode=os.O_RDONLY)
         self._img = converter.read_data(image)[0]  # can be RGB or greyscale
 
-        # we will fill the set of children with Components later in ._children
+        # we will fill the set of dependencies with Components later in ._dependencies
```

Comment:
No, from the code, it doesn't seem we fill the set of dependencies afterwards. You need to pass dependencies too here.